java/src/test/java/ai/rapids/cudf/TableTest.java

```diff
@@ -2085,6 +2084,116 @@ void testInnerJoinGatherMapsNulls() {
     }
   }
 
+  private void checkInnerDistinctJoin(Table leftKeys, Table rightKeys, Table expected,
+                                      boolean compareNullsEqual) {
+    GatherMap[] maps = leftKeys.innerDistinctJoinGatherMaps(rightKeys, compareNullsEqual);
+    try {
+      verifyJoinGatherMaps(maps, expected);
+    } finally {
+      for (GatherMap map : maps) {
+        map.close();
+      }
+    }
```

Comment:
Can we implement `GatherMap` as auto-closable?

Reply:
GatherMap is auto-closeable.  This is an array of GatherMaps.

Reply:
Hah, array of auto-closeable can't be auto-closeable. That sucks.